### PR TITLE
Mar 1118

### PR DESCRIPTION
**Pull request by jamespacileo**

### CODE REVIEW

File: api/barriers/models.py [hunk 1]
```diff
@@ -1625,3 +1632,91 @@ def status_filter(self, queryset, name, value):
...
+
+        recipient_emails = settings.SEARCH_DOWNLOAD_APPROVAL_REQUEST_EMAILS
+
+        client = NotificationsAPIClient(settings.NOTIFY_API_KEY)
+        group_id = Group.objects.get(
+            name=settings.APPROVED_FOR_BARRIER_DOWNLOADS_GROUP_NAME
+        ).id
+        user_group_approval_path = f"/users/add/?group={group_id}".format(
+            group_id=group_id
+        )
```
Comment (kevincarrogan) [inline api/barriers/models.py:1698]: Is this supposed to be an f-string and using format at the same time?

File: api/user/migrations/0024_create_action_plans_group.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    group.permissions.add(view_action_plans_permission)
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("user", "0023_create_barrier_search_download_approved_group"),
+    ]
+
+    operations = [migrations.RunPython(create_action_plans_group)]
```
Comment (kevincarrogan) [inline api/user/migrations/0024_create_action_plans_group.py:32]: Without specifying `reverse_code` this means that this migration is irreversible, is that what you meant for it to be?

File: api/user/serializers.py [hunk 3]
```diff
@@ -105,6 +108,15 @@ def get_permitted_applications(self, obj):
             return sso_me.get("permitted_applications", None)
         return None
 
+    def get_has_approved_digital_trade_email(self, obj):
+        email_domain = obj.email.split("@")[-1].lower()
+        print(
+            "settings.APPROVED_DIGITAL_TRADE_EMAILS",
+            settings.APPROVED_DIGITAL_TRADE_EMAIL_DOMAINS,
+            email_domain,
+        )
```
Comment (kevincarrogan) [inline api/user/serializers.py:117]: Remove the debug or make a logging call.

**Review by kevincarrogan — CHANGES_REQUESTED** [review]

File: api/user/migrations/0024_create_action_plans_group.py (hunk 2 above)
Comment (jamespacileo) [reply api/user/migrations/0024_create_action_plans_group.py:32]: added reverse code for the migration

File: api/barriers/models.py (hunk 1 above)
Comment (jamespacileo) [reply api/barriers/models.py:1698]: face palm moment, fixed to being just an f-string

File: api/user/serializers.py (hunk 3 above)
Comment (jamespacileo) [reply api/user/serializers.py:117]: removed

File: api/barriers/migrations/0114_auto_20211126_1422.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+            ),
+            preserve_default=False,
+        ),
+        migrations.AddField(
+            model_name="barriersearchcsvdownloadevent",
+            name="email",
+            field=models.EmailField(default="", max_length=254),
+            preserve_default=False,
+        ),
+        migrations.RunPython(remove_existing_download_entries),
```
Comment (kevincarrogan) [inline api/barriers/migrations/0114_auto_20211126_1422.py:44]: This doesn't have a `reverse_code` so isn't reversible.

Comment (jamespacileo) [reply api/barriers/migrations/0114_auto_20211126_1422.py]: added noop as reverse_code

**Review by kevincarrogan — APPROVED** [review]

**Pull request merged**